### PR TITLE
3295 add logging to api calls

### PR DESCRIPTION
**Pull request by Fbasham**
### Description
per title

### Related Azure Boards Work Items
[AB#3295](https://dev.azure.com/DTS-STN/1fc40a8f-28cf-47bc-b6e4-1c234bd06177/_workitems/edit/3295)

### Checklist

- [x] I have tested the changes locally
- [ ] I have updated the documentation if necessary
- [ ] I have added/updated tests that prove my fix is effective or that my feature works
- [x] I have checked that my code follows the project's coding style by running `npm run format:check`
- [x] I have checked that my code contains no linting errors by running `npm run lint`
- [x] I have checked that my code contains no type errors by running `npm run typecheck`
- [x] I have checked that all unit tests pass by running `npm run test:unit -- run`
- [x] I have checked that all e2e tests pass by running `npm run test:e2e`

### CODE REVIEW

File: frontend/app/services/address-service.server.ts [hunk 1]
```diff
@@ -37,11 +37,14 @@ function createAddressService() {
   const { INTEROP_API_BASE_URI } = getEnv();
 
   async function getAddressInfo(userId: string, addressId: string) {
+    log.trace('Calling getAddressInfo');
```
Comment (gregory-j-baker) [inline frontend/app/services/address-service.server.ts:40]: I think this should be logged at the DEBUG level, and should contain the user id and address ID.. something like:

```
log.debug('Fetcing address info [%s] for user [%s]', addressId, userId);
```

Think about when we're looking through the logs, trying to troubleshoot an address issue. Seeing "calling getaddressinfo" in the logs with no more information isn't useful.

Comment (gregory-j-baker) [reply frontend/app/services/address-service.server.ts:40]: Also, we generally wrap the values in some delimiter (I use `[..]`), so we can see if there are any extra spaces or other weird characters.

File: frontend/app/services/address-service.server.ts [hunk 2]
```diff
@@ -37,11 +37,14 @@ function createAddressService() {
...
     const url = `${INTEROP_API_BASE_URI}/users/${userId}/addresses/${addressId}`;
     const response = await fetch(url);
 
     if (response.ok) {
-      return toAddressInfo(addressDtoSchema.parse(await response.json()));
+      const data = await response.json();
+      log.trace('Method getAddressInfo returned: %j', data);
```
Comment (gregory-j-baker) [inline frontend/app/services/address-service.server.ts:46]:
```suggestion
      log.debug('Returning address [%s] for user [%s]', addressId, userId);
      log.trace('Address [%s]: [%j]', addressId, data);
```

File: frontend/app/services/address-service.server.ts [hunk 3]
```diff
@@ -37,11 +37,14 @@ function createAddressService() {
...
+      return toAddressInfo(addressDtoSchema.parse(data));
     }
 
     if (response.status === 404) {
```
Comment (gregory-j-baker) [inline frontend/app/services/address-service.server.ts:50]:
```
log.debug('Did not find address [%s] for user [%s]', addressId, userId);
```

**Review by gregory-j-baker — CHANGES_REQUESTED** [review]
Left a few comments that apply to the entire PR. It would take too long to make the same comment everywhere.

Let me know if you have any questions or if I can help out.

File: frontend/app/services/address-service.server.ts (hunk 1 above)
Comment (Fbasham) [reply frontend/app/services/address-service.server.ts]: Fair point. I've made changes accordingly.

**Review by gregory-j-baker — DISMISSED** [review]

**Review by gregory-j-baker — APPROVED** [review]

**Pull request merged**